packages/wrangler/templates/pages-template-plugin.ts

```diff
@@ -125,8 +125,12 @@ export default function (pluginArgs: unknown) {
 		const { env, next, data } = workerContext;
 
 		const url = new URL(request.url);
+		// TODO: Replace this with something actually legible. Should probably rely on the length of the `functionPath` rather than this split/join chaos.
 		const relativePathname = `/${
-			url.pathname.split(workerContext.functionPath)[1] || ""
+			url.pathname
+				.split(workerContext.functionPath)
+				.slice(1)
+				.join(workerContext.functionPath) || ""
```

Comment:
Are you just trying to replace the `functionPath` with an empty string `""` if it is found?

| `functionPath`  | `path`      |  `relativePath` |
|---|---|---|
| foo | /foo | / |
| foo | /foo/bar | /bar |
| foo | /bar/foo | /bar/foo |
| foo/bar | /foo | /foo |
| foo/bar | /foo/bar | / |
| foo/bar | /foo/bar/moo | /moo |
| "" | / | / |
| "" | /foo | /foo |

Reply:
If so can this just be done with a `.replace()` call?